example/22_cgemm/CMakeLists.txt

```diff
@@ -1 +1,11 @@
+add_custom_target(example_cgemm_xdl)
+
+add_example_executable_no_testing(example_cgemm_xdl_bf16 cgemm_xdl_bf16.cpp)
```

Comment:
We do not have basic operation for BF16, so bf16 verification will fail.
In pure gemm verification, we compare the gpu bf16 result with cpu fp32 result.
I thought we need to implement bf16 datatype in CK.
@asroy

Reply:
@rocking5566  I can't see in gemm example comparison with fp32: https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/develop/example/01_gemm/gemm_xdl_bf16.cpp#L224-L234. However I think that such comparison makes sense.

Reply:
I mean we should not simply using bhalf_t = ushort;
we need to implement like following.

class bhalf_t 
{
void operator*
....etc
}

The elder version of gemm_bf16 verification is like this 
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/d1db6a0c3ea190996bdae37adda191f746bfc34e/example/01_gemm/gemm_xdl_bf16.cpp#L216-L233

Current version will be incorrect for following parameter
./bin/example_gemm_xdl_bf16 1 2 0 1024 1024 1024 1024 1024 1024

Reply:
> I mean we should not simply using bhalf_t = ushort; we need to implement bf16 like following. https://github.com/ROCmSoftwarePlatform/MIOpen/blob/0183095329e7c1803e65de64350c7a057562d068/src/include/miopen/bfloat16.hpp#L32
> 

I'd recommend to reuse some version that has already wider community adoption. In. eg the one from [Eigen](https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Core/arch/Default/BFloat16.h) is very popular. Then all(most) you need to do is just expose this type like this https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/bfloat16.h#L24-L26.

Reply:
hip-clang doesn't natively support bfloat16 on either GPU or CPU, it treat bfloat16 as uint16. That's why CK does aliasing.

Having CK creating an wrapper for it is a real manintaince pain. Proper way to fix it is for compiler to add a new datatype for bfloa16 instead of aliasing.